runners/java-fn-execution/src/test/java/org/apache/beam/runners/fnexecution/environment/ProcessManagerTest.java

```diff
@@ -99,4 +106,53 @@ public void testEnvironmentVariables() throws IOException, InterruptedException
     assertThat(process.getUnderlyingProcess().exitValue(), is(1));
     processManager.stopProcess("1");
   }
+
+  @Test
+  public void testRedirectOutput() throws IOException, InterruptedException {
+    File outputFile = File.createTempFile("beam-redirect-output-", "");
+    outputFile.deleteOnExit();
+    ProcessManager processManager = ProcessManager.create();
+    ProcessManager.RunningProcess process =
+        processManager.startProcess(
+            "1",
+            "bash",
+            Arrays.asList("-c", "echo 'testing123'"),
+            Collections.emptyMap(),
+            outputFile);
+    for (int i = 0; i < 10 && process.getUnderlyingProcess().isAlive(); i++) {
+      Thread.sleep(100);
+    }
+    processManager.stopProcess("1");
+    byte[] output = Files.readAllBytes(outputFile.toPath());
+    assertNotNull(output);
+    String outputStr = new String(output, Charset.defaultCharset());
+    assertThat(outputStr, containsString("testing123"));
+  }
+
+  @Ignore("not working yet")
+  @Test
+  public void testInheritIO() throws IOException, InterruptedException {
```

Comment:
@mxm I see the redirect working as the expected output is printed in the console. But short of starting a separate java process I don't see a way to capture it in the test.

Reply:
I left the test but skipped the assertion, maybe we can still find a clever way to do this.